vector_double.cpp

```diff
@@ -0,0 +1,207 @@
+#include "vector_double.hpp"
+
+
+template <typename T>
+Vector<T>::Vector()
+{
+    size_v = 0;
+    space  = 0;
+    elem   = nullptr;
+
+}
+
+
+// Copy Constructor
+template<class T>
+Vector<T>::Vector(const Vector<T>& b)
+{
+    space = b.space;
+    size_v = b.size_v;
+    elem = b.elem;
+
+}
+template<class T>
+Vector<T>::Vector(Vector<T>&& obj)
+{
+    space = obj.space;
+    size_v = obj.size_v;
+    elem = obj.elem;
+}
+
+//explicit
+template<class T>
+Vector<T>::Vector(unsigned int)
+{
+    size_v = 0;
+    space  = 0;
+    elem   = nullptr;
+}
+
+
+template<class T>
+Vector<T>::~Vector()
+{
+    delete [] elem;
+}
+
+template<class T>
+int Vector<T>:: size() const
+{
+    return size_v;
+}
+
+template<class T>
+int Vector<T>:: capacity()const
+{
+    return space;
+}
+
+template<typename T>
+void Vector<T>::push_back(const T &v)
+{
+
+    if (size_v >= space)
+    {
+        reserve(space +5);
+        elem [size_v++] = v;
+    }
+
+}
+
+template<class T>
+void Vector<T>::pop_back()
+{
+    if(size_v > 0)
+    {
+        elem[size_v -1];
+    }
+}
+
+
+template<class T>
+void Vector<T>::resize(unsigned int size)
+{
+
+    reserve(size);
+    size_v = size;
+}
+
+
+template<class T>
+void Vector<T>::reserve(unsigned int capacity)
+{
+    unsigned int l_Size;
+
+    if(elem == 0)
+    {
+        size_v = 0;
+        space = 0;
+    }
+    T * newElem = new T [space];
+
+    l_Size = space < size_v ? capacity : size_v;
+
+    //copy (buffer, buffer + l_Size, newElem);
+    for (unsigned int i = 0; i < l_Size; i++)
+        newElem[i] = elem[i];
+
+    space = capacity;
+    delete[] elem;
+    elem = newElem;
+}
+
+
+template<class T>
+void Vector<T>::clear()
+{
+    space = 0;
+    size_v = 0;
+    elem = 0;
+}
+
+template<class T>
+Vector<T> Vector<T>::operator=(Vector && other)
+{
+     move(other);
+     return *this;
+}
+
+
+template<class T>
+Vector<T> Vector<T>::operator=(const Vector &v)
+{
+    delete[ ] elem;
+    size_v = v.size_v;
+    space = v.space;
+    elem = new T [size_v];
+    for (int i = 0; i < size_v; i++)
+    {
+        elem[i] = v.elem[i];
+    }
+    return *this;
+}
+
+template <class T>
+T& Vector<T>::operator[](unsigned int index)
+{
+    return elem[index];
+}
+
+template <class T>
+T* Vector<T>::begin()
```

Comment:
so you defined `iterator` but aren't using it? I think begin() and end() should return the canonical iterator type for clarity